### ISSUE TITLE
hangs on save

### ISSUE THREAD

**Issue by nschloe** [neovim/neovim#3449]
Since a few days, I'm experiencing hangs when saving a file in nvim:

```
$ ps aux | grep nvim
nschloe   6023  0.0  0.2 172412 17524 ?        Sl   12:18   0:00 nvim ../matplotlib2tikz.py
nschloe  11593  0.5  0.2 172412 17508 ?        Sl   14:20   0:00 nvim setup.p
```

This is with

```
$ nvim --version
NVIM 0.0.0-alpha (compiled Oct 12 2015 02:36:33)
Commit: ?
Build type: RelWithDebInfo
Compilation: /usr/bin/x86_64-linux-gnu-gcc -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -I/build/neovim-0qd3aW/neovim-0.0.0+git201510120122+1932+14~ubuntu15.04.1/build/config -I/build/neovim-0qd3aW/neovim-0.0.0+git201510120122+1932+14~ubuntu15.04.1/src -I/build/neovim-0qd3aW/neovim-0.0.0+git201510120122+1932+14~ubuntu15.04.1/.deps/usr/include -I/build/neovim-0qd3aW/neovim-0.0.0+git201510120122+1932+14~ubuntu15.04.1/.deps/usr/include -I/usr/include/luajit-2.0 -I/build/neovim-0qd3aW/neovim-0.0.0+git201510120122+1932+14~ubuntu15.04.1/.deps/usr/include -I/build/neovim-0qd3aW/neovim-0.0.0+git201510120122+1932+14~ubuntu15.04.1/.deps/usr/include -I/build/neovim-0qd3aW/neovim-0.0.0+git201510120122+1932+14~ubuntu15.04.1/.deps/usr/include -I/build/neovim-0qd3aW/neovim-0.0.0+git201510120122+1932+14~ubuntu15.04.1/.deps/usr/include -I/usr/include -I/build/neovim-0qd3aW/neovim-0.0.0+git201510120122+1932+14~ubuntu15.04.1/build/src/nvim/auto -I/build/neovim-0qd3aW/neovim-0.0.0+git201510120122+1932+14~ubuntu15.04.1/build/include
Compiled by buildd@lgw01-48

Optional features included (+) or not (-): +acl   +iconv    +jemalloc 
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/nvimrc"
     user vimrc file: "~/.nvimrc"
 2nd user vimrc file: "~/.nvim/nvimrc"
      user exrc file: "~/.exrc"
  fall-back for $VIM: "/usr/share/nvim"
```

What can I do to debug?

**Comment by teto** [comment]
Do you experience the same slowdown without the plugins ?

**Comment by nschloe** [comment]
Haven't tried since my workflow depends on the plugins and the hangs only occur once in a while. Is there a debug log that I could consult?

**Comment by myitcv** [comment]
I sporadically have the same problem. Just created a core dump if that's useful?

```
NVIM 0.0.0-alpha+201510160154 (compiled Oct 16 2015 16:21:34)
Commit: 3a970e57dfd48f090f8ccc21567b7974e13d4c68
```

**Comment by dgryski** [comment]
I think this is a more generic "hang on exit", rather than "hang on save".  I reproduced it under gdb, and got the following backtrace:
```
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff79bb8ed in pthread_join (threadid=140737312192976, thread_return=0x0) at pthread_join.c:90
90	pthread_join.c: No such file or directory.
(gdb) 
(gdb) 
(gdb) 
(gdb) bt
#0  0x00007ffff79bb8ed in pthread_join (threadid=140737312192976, thread_return=0x0) at pthread_join.c:90
#1  0x00000000005ab1fe in uv_thread_join ()
#2  0x0000000000586439 in ?? ()
#3  0x0000000000538263 in ui_builtin_stop ()
#4  0x0000000000556010 in mch_exit ()
#5  0x0000000000559517 in getout ()
#6  0x00000000005597ec in ?? ()
#7  0x000000000055a076 in main ()
```

**Comment by tarruda** [comment]
Seems like a deadlock. It will be fixed after 0.1

**Comment by justinmk** [comment]
If this was a hang-on-exit it may be helped by 68bcb32ec43e2fab30dc05439fc77cf28793922c. Otherwise assuming it is covered by the 0.1 changes mentioned.
